### ISSUE TITLE
Redis 2.4.0 hangs when getting a key

### ISSUE THREAD

**Issue by cressie176** [NodeRedis/node_redis#924]
```
    var client = redis.createClient('redis://redis-postcodes.somehost.com:6379');
    console.log('Connecting to ', client.address)
    client.get(postcode, function(err, result){
        console.log('Connected to', client.address);
        client.end()
        return cb(err, result)
    })
```

Output with 2.3.1
```
Connecting to  redis-postcodes.somehost.com:6379
Connected to redis-postcodes.somehost.com:6379
```

Output with 2.4.0
```
Connecting to  redis-postcodes.somehost.com:6379
```

The redis service is an AWS ElastiCache 2.8.22

**Comment by BridgeAR** [comment]
I noticed this myself about one hour after releasing 2.4 :(

I'm already investigating the issue and try to fix it. If I do not find a fix in the next couple of hours, I'll revert the parser changes. Please use the hiredis parser in the meanwhile or use 2.3 until I release 2.4.1.

**Comment by cressie176** [comment]
Thanks. No problem using 2.3

**Comment by BridgeAR** [comment]
@cressie176 would you be so kind and test my fix: c782ef287f225938abb65d7e41ee0928206ef17a

**Comment by cressie176** [comment]
Works for me. Thanks.
